cdk/src/functions/service-control-policies-root.ts

```diff
@@ -0,0 +1,137 @@
+import {
+  AttachPolicyCommand,
+  CreatePolicyCommand,
+  DeletePolicyCommand,
+  DetachPolicyCommand,
+  ListRootsCommand,
+  OrganizationsClient,
+  PolicyType,
+  UpdatePolicyCommand,
+  ListPoliciesCommand,
+} from '@aws-sdk/client-organizations';
+import { CdkCustomResourceEvent, CdkCustomResourceResponse, Context } from 'aws-lambda';
+
+async function getRootId(organizationClient: OrganizationsClient): Promise<string | undefined> {
+  let id = 'error'; //set to error. Update if root ID is found. Then, update it to Root Id.
+  const command = new ListRootsCommand({});
+  const response = await organizationClient.send(command);
+
+  if (!response.Roots || response.Roots.length === 0) {
+    console.warn('No roots found in the organization');
+    return id;
+  }
+
+  const root = response.Roots[0];
+  id = root.Id || 'error';
+  return id;
+}
+
+async function getPolicyId(organizationClient: OrganizationsClient, policyName: string) {
+  const commandListPolicies = new ListPoliciesCommand({
+    Filter: PolicyType.SERVICE_CONTROL_POLICY,
+  });
+
+  const response = await organizationClient.send(commandListPolicies);
+
+  let policyId = 'error'; //set to error. Update if superwerker-root SCP policy is found. Then, update it to SCP Policy Id.
+
+  response.Policies?.forEach((policy) => {
+    if (policy.Name === policyName) {
+      policyId = policy.Id;
+    }
+  });
+
+  return policyId;
+}
+
+export async function handler(event: CdkCustomResourceEvent, _context: Context): Promise<CdkCustomResourceResponse> {
+  let client = new OrganizationsClient({ region: 'us-east-1' });
+
+  switch (event.RequestType) {
+    case 'Create':
+      console.log('Creating Policy for : ', event.LogicalResourceId);
+      try {
+        let rootId = await getRootId(client);
+
+        const commandCreatePolicy = new CreatePolicyCommand({
+          Type: PolicyType.SERVICE_CONTROL_POLICY,
+          Description: `superwerker - ${event.LogicalResourceId}`,
+          Name: event.ResourceProperties.scpName,
+          Content: event.ResourceProperties.policy,
+        });
+
+        const responseCreatePolicy = await client.send(commandCreatePolicy);
+
+        const commandAttachPolicy = new AttachPolicyCommand({
+          PolicyId: responseCreatePolicy.Policy?.PolicySummary?.Id || '',
+          TargetId: rootId,
+        });
+
+        await client.send(commandAttachPolicy);
+
+        return { SUCCESS: 'SCPs have been successfully created for Root account' };
+      } catch (e) {
+        console.log('Error during Creating Policy: ', e);
+        return { ErrorMessage: `Error during creating policy: ${e}` };
```

Comment:
in the old code there was no catch all. I think we should fail execution for most of the errors.